InformationScripting/src/queries/QueryExecutor.cpp

```diff
@@ -57,11 +67,17 @@ Interaction::CommandResult* QueryExecutor::execute()
 		}
 		else
 		{
-			return new Interaction::CommandResult(new Interaction::CommandError(results[0].errors()[0]));
+			hasError = true;
+			errorMessage = results[0].errors()[0];
 		}
 	}
 
-	return new Interaction::CommandResult();
+	if (queries_.empty()) delete this;
```

Comment:
I think deleteLater is just:
``` QCoreApplication::postEvent(this, new QDeferredDeleteEvent());```

see also: https://github.com/qtproject/qtbase/blob/dev/src/corelib/kernel/qobject.cpp#L2136
so I think I can use this.